vignettes/DBI.Rmd

```diff
@@ -13,109 +13,183 @@ vignette: >
-This DBMS-agnostic approach is possible, because DBI works best with several other packages that act as drivers to absorb the peculiarities of the specific DBMSs.
+The {DBI} (**D**ata**B**ase **I**nterface) package provides a simple, consistent interface between R and database management systems (DBMS).  
+Each supported DBMS is supported by its own R package that implements the [DBI specification](https://cran.r-project.org/web/packages/DBI/vignettes/spec.html). You may need to install the package specific to your DBMS.
 
-These packages import {DBI} and implement its methods depending on the specific database management system.
-
-Currently, DBI works with the [many different database management systems](https://github.com/r-dbi/DBI/issues/274), e.g.:
+DBI currently supports 30 DBMS, including:
 
 * MySQL, using the R-package [RMySQL](https://github.com/r-dbi/RMySQL)
 * MariaDB, using the R-package [RMariaDB](https://github.com/r-dbi/RMariaDB)
-* Postgres, using and the R-package [RPostgres](https://github.com/r-dbi/RPostgres)
-* SQLite, using and the R-package [RSQLite](https://github.com/r-dbi/RSQLite)
+* Postgres, using the R-package [RPostgres](https://github.com/r-dbi/RPostgres)
+* SQLite, using the R-package [RSQLite](https://github.com/r-dbi/RSQLite)
 
-DBI offers a set of classes and methods that define what operations are possible and how they are performed:
+For the full list of supported DBMS visit [https://github.com/r-dbi/backends](https://github.com/r-dbi/backends)
 
-* connect/disconnect to the DBMS
-* create and execute statements in the DBMS
-* extract results/output from statements
-* error/exception handling
-* information (meta-data) from database objects
-* transaction management (optional)
+The functionality currently supported for each of these DBMS's includes:
+- [manage a connection to a database](#connecting-to-a-database)
+- list the tables in a database [link]
+- get table details/column headers/fields [link]
+- read a table into a dataframe [link]
+- read a portion of a table into a dataframe [link]
+- run SQL queries, including parameterised queries [link]
 
-## Examples
+## How to connect to a database using DBI
 
-To showcase DBI capabilities, we create a in-memory RSQLite database 
+Connections to databases are created using the `dbConnect()` function. 
+The first argument to the function is the driver for the DBMS you are connecting to. 
+In the example below we are connecting to a MariaDB instance, so we use the RMariaDB::MariaDB() driver. 
+The rest of the arguments depend on the authentication required by the DBMS. In the example username, password, host, port, dbname are required. 
+See the documentation for the DBMS driver package that you are using for specifics.
+The databse connection will need to be closed at the end of your session. This is covered in the section [How to close a database connection](#how-to-close-a-database-connection).
 
```

Comment:
Can you explain a bit what this data source is and what data we'll be looking at? Is this the best dataset for this tutorial?